stock_move_packaging_qty/models/stock_move.py

```diff
@@ -0,0 +1,111 @@
+            if (
+                not move.product_packaging_id
+                or move.product_qty == 0
+                or move.product_packaging_id.qty == 0
+            ):
+                move.product_packaging_qty = 0
+                continue
+            # Consider uom
+            if move.product_id.uom_id != move.product_uom:
+                product_qty = move.product_uom._compute_quantity(
+                    move.product_uom_qty, move.product_id.uom_id
+                )
+            else:
+                product_qty = move.product_uom_qty
+            move.product_packaging_qty = product_qty / move.product_packaging_id.qty
+
+    def _prepare_product_packaging_qty_values(self):
+        return {
+            "product_uom_qty": self.product_packaging_id.qty
+            * self.product_packaging_qty,
+            "product_uom": self.product_packaging_id.product_uom_id.id,
+        }
+
+    def _inverse_product_packaging_qty(self):
+        for move in self:
+            if move.product_packaging_qty and not move.product_packaging_id:
+                raise UserError(
+                    _(
+                        "You must define a package before setting a quantity "
+                        "of said package."
+                    )
+                )
+            if move.product_packaging_id and move.product_packaging_id.qty == 0:
+                raise UserError(
+                    _("Please select a packaging with a quantity bigger than 0")
+                )
+            if move.product_packaging_id and move.product_packaging_qty:
+                move.write(move._prepare_product_packaging_qty_values())
+
+    @api.onchange("product_packaging_id")
+    def _onchange_product_packaging(self):
+        if self.product_packaging_id:
+            self.update(
+                {
+                    "product_packaging_qty": 1,
+                    "product_uom_qty": self.product_packaging_id.qty,
+                    "product_uom": self.product_id.uom_id,
+                }
+            )
+        else:
+            self.update({"product_packaging_qty": 0})
+        if self.product_packaging_id:
+            return self._check_package()
+
+    @api.onchange("product_packaging_qty")
+    def _onchange_product_packaging_qty(self):
+        if self.product_packaging_qty and self.product_packaging_id:
+            self.update(self._prepare_product_packaging_qty_values())
+
+    @api.onchange("product_uom_qty", "product_uom")
+    def onchange_quantity(self):
+        res = super().onchange_quantity()
+        if not res:
+            res = self._check_package()
+        return res
+
+    def _check_package(self):
+        default_uom = self.product_id.uom_id
+        pack = self.product_packaging_id
+        qty = self.product_qty
+        q = default_uom._compute_quantity(pack.qty, self.product_uom)
+        if qty and q and round(qty % q, 2):
+            newqty = qty - (qty % q) + q
+            return {
```

Comment:
I would have changed this behavior as onchanges become more and more obsolete.

Reply:
Makes sense, but I prefer to keep the migration diff smaller.

Reply:
Ok, so, at least put it in ROADMAP.rst